src/client/Microsoft.Identity.Client/ManagedIdentity/KeyProviders/WindowsCngKeyOperations.cs

```
@@ -0,0 +1,142 @@
+                {
+                    key = CngKey.Open(
+                        KeyGuardKeyName,
+                        new CngProvider(SoftwareKspName),
+                        CngKeyOpenOptions.UserKey | CngKeyOpenOptions.Silent);
+                }
+                catch (CryptographicException)
+                {
+                    // Not found -> create fresh (helper may return null if VBS unavailable)
+                    logger?.Verbose(() => "[MI][WinKeyProvider] KeyGuard key not found; creating fresh.");
+                    key = KeyGuardHelper.CreateFresh(logger);
+                }
+
+                // If VBS is unavailable, CreateFresh() returns null. Bail out cleanly.
+                if (key == null)
+                {
+                    logger?.Verbose(() => "[MI][WinKeyProvider] KeyGuard unavailable (VBS off or not supported).");
+                    return false;
+                }
+
+                // Ensure actually KeyGuard-protected; recreate if not
+                if (!KeyGuardHelper.IsKeyGuardProtected(key))
+                {
+                    logger?.Verbose(() => "[MI][WinKeyProvider] KeyGuard key found but not protected; recreating.");
+                    key.Dispose();
+                    key = KeyGuardHelper.CreateFresh(logger);
+
+                    // Check again after recreate; still null or not protected -> give up KeyGuard path
+                    if (key == null || !KeyGuardHelper.IsKeyGuardProtected(key))
+                    {
+                        key?.Dispose();
+                        logger?.Verbose(() => "[MI][WinKeyProvider] Unable to obtain a KeyGuard-protected key.");
+                        return false;
+                    }
+                }
+
+                rsa = new RSACng(key);
+                if (rsa.KeySize < 2048)
+                {
+                    try
+                    { rsa.KeySize = 2048; }
+                    catch { /* some providers don't allow */ }
```

Why would we fail silently if we attempt to extend the key size and it fails? Doesnt it create a false sense of security?

---

Let's log the error? In any case there will be an attestation token involved that proves the key is in hardware.